File: Lista de Exercicio Paradigmas/Lista Java/Ex11.java
abstract class Funcionario {
    abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private int horasTrabalhadas;
    private double valorHora;

    public FuncionarioHorista(int horasTrabalhadas, double valorHora) {
        this.horasTrabalhadas = horasTrabalhadas;
        this.valorHora = valorHora;
    }

    @Override
    double calcularSalario() {
        return horasTrabalhadas * valorHora;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salarioMensal;

    public FuncionarioAssalariado(double salarioMensal) {
        this.salarioMensal = salarioMensal;
    }

    @Override
    double calcularSalario() {
        return salarioMensal;
    }
}

public class Main {
    public static void main(String[] args) {
        FuncionarioHorista horista = new FuncionarioHorista(160, 20);
        FuncionarioAssalariado assalariado = new FuncionarioAssalariado(3000);

        System.out.println("Salário do Funcionário Horista: R$ " + horista.calcularSalario());
        System.out.println("Salário do Funcionário Assalariado: R$ " + assalariado.calcularSalario());
    }
}

File: Lista de Exercicio Paradigmas/Lista Python/Ex11.py
from abc import ABC, abstractmethod

# Classe abstrata Funcionario
class Funcionario(ABC):
    @abstractmethod
    def calcularSalario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas_trabalhadas, valor_hora):
        self.horas_trabalhadas = horas_trabalhadas
        self.valor_hora = valor_hora

    def calcularSalario(self):
        return self.horas_trabalhadas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_mensal):
        self.salario_mensal = salario_mensal

    def calcularSalario(self):
        return self.salario_mensal

horista = FuncionarioHorista(160, 20) 
assalariado = FuncionarioAssalariado(3000)  
print(f'Salário do Funcionário Horista: R$ {horista.calcularSalario()}')
print(f'Salário do Funcionário Assalariado: R$ {assalariado.calcularSalario()}')
